### PR TITLE
Phpstan lvl 7

### PR DESCRIPTION
**Pull request by greg0ire**

### CODE REVIEW

**Comment by greg0ire** [comment]
Let's wait for https://www.githubstatus.com/incidents/zbpwygxwb3gw to be resolved before merging this.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# Circular dependency
+
+This package has a development dependency on `doctrine/common`, which has a
```
Comment (alcaeus) [inline CONTRIBUTING.md:3]: Thanks for documenting this. Is there anything we can do to fix this? Should we prepare doctrine/common to work with persistence 3?

**Review by alcaeus — APPROVED** [review]
Can't comment on the impact of the psalm changes, but in general LGTM.

File: CONTRIBUTING.md (hunk 1 above)
Comment (greg0ire) [reply CONTRIBUTING.md:3]: We could use `^2.0 || ^3.0`, but IMO that's a bit "risky" if we do not test that 2.0 actually work with 3, and the benefit is small: that would only fix one weird thing out of the 2 we have to do, namely:
- using `COMPOSER_ROOT_VERSION`
- using `2.0` as a value for it when developing on 3.x

**Pull request merged**